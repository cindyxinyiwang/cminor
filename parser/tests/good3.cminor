// statements

b: function void () = {
	// function declaration

	a: function integer (x: integer, y: boolean) = {
		print "hello", "world";
		return y;
	}
	// declaration
	
	x: integer = 20;
	// if statement

	if (x > 0) {
		m   = true;
	} else {
		x = x+1;
		y++;
	}
	// if statement
	
	if (x*3 <= 12) {
		print "this is awesome", x, 12;
	}
	// for 
	for (x =0; x < 12; x++) {
		m = m+n;
	}
	for (;;x++) {
		m = m+1;
	}
	return true;
	{
		x = x+1;
		y ++;
	}
}