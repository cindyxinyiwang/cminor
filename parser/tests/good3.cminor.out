b: function void () = {
	// function declaration

	a: function integer (x: integer, y: boolean) = {
		print "hello", "world";
		return y;
	}
	// declaration
	
	x: integer = 20;
	// if statement

	if (x > 0) {
		m   = true;
	} else {
		x = x+1;
		y++;
	}
	// if statement
	
	if (x*3 <= 12) {
		print "this is awesome", x, 12;
	}
	// for 
	for (x =0; x < 12; x++) {
		m = m+n;
	}
	for (;;x++) {
		m = m+1;
	}
	return true;
	{
		x = x+1;
		y ++;
	}
} : function void()= {
	a: function integer (x: integer, y: boolean) = {
		print "hello", "world";
		return y;
	} : function integer(x: integer, y: boolean) = {
		print "hello", "world";
		return y;
	}
	// declaration
	
	x: integer = 20;
	// if statement

	if (x > 0) {
		m   = true;
	} else {
		x = x+1;
		y++;
	}
	// if statement
	
	if (x*3 <= 12) {
		print "this is awesome", x, 12;
	}
	// for 
	for (x =0; x < 12; x++) {
		m = m+n;
	}
	for (;;x++) {
		m = m+1;
	}
	return true;
	{
		x = x+1;
		y ++;
	}
}: integer,y: boolean) = {
		print "hello", "world";
		return y;
	}
	// declaration
	
	x: integer = 20;
	// if statement

	if (x > 0) {
		m   = true;
	} else {
		x = x+1;
		y++;
	}
	// if statement
	
	if (x*3 <= 12) {
		print "this is awesome", x, 12;
	}
	// for 
	for (x =0; x < 12; x++) {
		m = m+n;
	}
	for (;;x++) {
		m = m+1;
	}
	return true;
	{
		x = x+1;
		y ++;
	}
}: boolean)= {
		print "hello","world";
		return y;;
	}
	x: integer = 20; : integer=20;
	if ((x >>0))
	{
		(m   ==true);
	}
	else 
	{
		(x ==(x++1));
		(y++++);
	}
	if (((x**3)<=12))
	{
		print "this is awesome",x,,12;
	}
	for ((x ==0);(x <<12);(x++++))
	{
		(m ==(m++n;));
	}
	for (;;(x++++))
	{
		(m ==(m++1));
	}
	return true;
	{
		(x ==(x++1));
		(y ++++);
	}
}
